src/app/routes/onDemandRadio/getInitialData/logInitialData/index.js

```diff
@@ -0,0 +1,18 @@
+import path from 'ramda/src/path';
+import nodeLogger from '#lib/logger.node';
+import { OD_RADIO_EPISODE_EXPIRED } from '#lib/logger.const';
+
+const logger = nodeLogger(__filename);
+
+const getUri = pageData => {
```

Comment:
I think it might be friendlier to work with if we could keep all the actual field retrieval logic in the `getInitialData` file if possible, in the same way we do now.  Having to go into this extra file to see where the data comes from might get a bit annoying

I think it makes sense to have this file for various helper functions and the like though - things that don't need to be changed often

For visibility, by PM I'd suggested we consider an API that works kind of like this:

![Screen Shot 2020-06-03 at 11 23 56](https://user-images.githubusercontent.com/8601460/83876454-cabeeb80-a730-11ea-8415-5593f980ea9b.png)

Curious what others think though

Reply:
We spoke about this on Slack but just to make public here for anyone else reading - the `getUri` function is only used by the logger, it isn't used as part of the field retrieval logic within `getInitialData`.

The API screenshot of the `pathWithLogging` idea Ryan has will be explored as part of https://github.com/bbc/simorgh/issues/6770